iotilecore/iotile/core/dev/annotated_registry.py

```diff
@@ -80,6 +81,15 @@ def list_config(self):
 
         return self.reg.list_config()
 
+    @param("device_uuid", "int", desc="Device UUID")
+    @param("password", "str", desc="Password to the device")
+    def set_temporary_password(self, device_uuid, password: str):
+        InMemoryAuthProvider.add_password(device_uuid, password)
+
+    @param("device_uuid", "int", desc="Device UUID")
+    def clear_temporary_password(self, device_uuid: int):
+        InMemoryAuthProvider.clear_password(device_uuid)
+
```

Comment:
Allows user to set a temporary password to connect to a device within the same hardware instance.